### DISCUSSION TITLE
Enhancing Search Grand Exchange alt-click option RFC

### DISCUSSION THREAD

**Discussion by deadmeu** [runelite/runelite#15908] [Ideas]
I think it would be nice if the "Search Grand Exchange" alt-click option would automatically remove the "(Members)" suffix on member's items when you are not in a members world.

I've got two working solutions for this but I'm creating this discussion thread as an RFC and would like some input as to which direction would be preferred (if any).

### Solution 1 - Manually strip out the members suffix
In [`GrandExchangeInputListener`](https://github.com/runelite/runelite/blob/master/runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeInputListener.java#L61) we can manually strip out the member's suffix:
``` Java
private static final String MEMBERS_SUFFIX = " (Members)";
...

String itemName = Text.removeTags(menuEntry.getTarget());
if (itemName.endsWith(MEMBERS_SUFFIX)) {
    itemName = itemName.substring(0, itemName.lastIndexOf(MEMBERS_SUFFIX));
}

search(itemName);
...
```

### Solution 2 - Get the members name from ItemComposition
Instead of manually stripping out the suffix, we could instead call the [`getMembersName`](https://github.com/runelite/runelite/blob/master/runelite-api/src/main/java/net/runelite/api/ItemComposition.java#L49) method on the item composition:
``` Java
@Inject
private ItemManager itemManager;
...

int widgetIndex = menuEntry.getParam0();
int widgetID = menuEntry.getParam1();

Widget w = getWidget(widgetID, widgetIndex);
int id = itemManager.canonicalize(w.getItemId());
String itemName = itemManager.getItemComposition(id).getMembersName();

search(itemName);
...
```

I think the second approach is nicer since we don't have to do any string manipulation. I'm new to Runelite development so there may be cleaner ways to do this - I'd love to know.

Alternatively, maybe [`GrandExchangeSearchPanel`](https://github.com/runelite/runelite/blob/master/runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeSearchPanel.java#L150) should handle this itself? The panel could accept items with the members suffix and automatically strip them out during the lookup.

Also, it would be nice to extend this enhancement to include repairable items, such as degraded Barrow's armour. Currently, searching against something like a Dharok's Platebody (50) doesn't yield any results.

**Comment by Adam-** [comment]
The second option is better, and you can just use `menuEntry.getWidget()` instead

**Reply by deadmeu** [reply]
Thanks! Any thoughts on degraded armour issue? If you're happy with it, do you know if there's an easy way to convert from a degraded, untradeable item id to its tradeable version?

**Reply by Alexsuperfly** [reply]
look at how https://github.com/runelite/runelite/blob/master/runelite-client/src/main/java/net/runelite/client/game/ItemMapping.java is used

**Reply by Adam-** [reply]
`ItemMapping` might do what you want, which is how item manager is able to get prices of those currently

**Reply by deadmeu** [reply]
I'm finding that `menuEntry.getWidget()` is returning `null`. Is there some sort of setup required?
![image](https://user-images.githubusercontent.com/12111013/201375887-1adc3cb7-1b78-460a-8fff-4e3a67df03e1.png)

**Reply by Adam-** [reply]
If the menu entry is of type RUNELITE then it won't work.
